src/realtime/realtime-hapi-plugin.ts

```diff
@@ -66,36 +68,29 @@ import {
 
 export const PING_INTERVAL = 20000;
 
+type StreamFactory = (request: Hapi.Request) => Promise<Observable<PersistedEvent<any>>>;
+
 @injectable()
-export class RealtimeHapiPlugin {
+export class RealtimeHapiPlugin extends HapiPlugin {
 
   public static injectSymbol = Symbol('realtime-plugin');
-  private jsonApiPlugin: JsonApiHapiPlugin;
-  private eventBus: PersistentEventBus;
   private eventBusSubscription: Subscription;
   public readonly persistedEvents: Observable<PersistedEvent<any>>;
-  private readonly logger: logger.Logger;
 
   constructor(
-    @inject(JsonApiHapiPlugin.injectSymbol) jsonApiPlugin: JsonApiHapiPlugin,
-    @inject(eventBusInjectSymbol) eventBus: PersistentEventBus,
-    @inject(logger.loggerInjectSymbol) logger: logger.Logger) {
-
-    this.eventBus = eventBus;
-    this.logger = logger;
-    this.jsonApiPlugin = jsonApiPlugin;
+    @inject(JsonApiHapiPlugin.injectSymbol) private readonly jsonApiPlugin: JsonApiHapiPlugin,
+    @inject(eventBusInjectSymbol) private readonly eventBus: PersistentEventBus,
+    @inject(logger.loggerInjectSymbol) private readonly logger: logger.Logger,
+  ) {
+    super({
+        name: 'realtime-plugin',
+        version: '1.0.0',
+    });
     this.persistedEvents = this.eventBus.getStream()
       .filter(isPersistedEvent)
       .map(event => <PersistedEvent<any>> event)
```

Comment:
It's purpose is only to make the compiler agree that `this.persistedEvents` has the desired narrowed type.